## ESTIMATION AND CONFIDENCE INTERVALS

In [ ]:
### Background
### Manufacturer of print-heads is interested in the mean durability (in millions
### of characters printed before failure) of their print-heads.
### Destructive testing -> small sample size due to cost constraints.
###
### Data: durability of 15 print-heads (millions of characters)
### 1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48,
### 1.20, 1.33, 1.18, 1.22, 1.29
### n = 15

### Part (a): 99% Confidence Interval Using the Sample Standard Deviation

**Step 1: Why the t-distribution?**

The population standard deviation (σ) is **not known** here — we only have the sample data. Whenever σ is unknown and it must be estimated from the sample itself (using the sample standard deviation, s), the extra uncertainty this introduces means the sampling distribution of the mean is no longer exactly Normal — it follows a **t-distribution** with (n − 1) degrees of freedom. The t-distribution has heavier tails than the Normal distribution, which widens the interval slightly to account for the added uncertainty of estimating σ from a small sample (n = 15 < 30).

**Step 2: Compute the sample statistics**

- Sample mean: x̄ = Σx / n
- Sample standard deviation: s = √[ Σ(x − x̄)² / (n − 1) ]

**Step 3: Formula for the Confidence Interval**

CI = x̄ ± t(α/2, n−1) × (s / √n)

where t(α/2, n−1) is the critical t-value for a 99% confidence level (α = 0.01) with (n − 1) = 14 degrees of freedom.

In [1]:
# Python Code — Part (a)
import numpy as np
from scipy import stats

data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07,
                 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

n = len(data)
x_bar = np.mean(data)
s = np.std(data, ddof=1)          # sample standard deviation

confidence = 0.99
alpha = 1 - confidence
df = n - 1

t_crit = stats.t.ppf(1 - alpha/2, df)
se = s / np.sqrt(n)
margin_error_t = t_crit * se

ci_lower_t = x_bar - margin_error_t
ci_upper_t = x_bar + margin_error_t

print("Sample size (n):", n)
print("Sample mean (x̄):", round(x_bar, 4))
print("Sample std dev (s):", round(s, 4))
print("Degrees of freedom:", df)
print("Critical t-value:", round(t_crit, 4))
print("Margin of Error:", round(margin_error_t, 4))
print("99% Confidence Interval (t-distribution): (", round(ci_lower_t, 4), ",", round(ci_upper_t, 4), ")")

Sample size (n): 15
Sample mean (x̄): 1.2387
Sample std dev (s): 0.1932
Degrees of freedom: 14
Critical t-value: 2.9768
Margin of Error: 0.1485
99% Confidence Interval (t-distribution): ( 1.0902 , 1.3871 )


**Conclusion (Part a):**
We are 99% confident that the true mean number of characters printed before print-head failure lies between approximately **1.090 million and 1.387 million characters**, based on the sample standard deviation and the t-distribution.

### Part (b): 99% Confidence Interval Using the Known Population Standard Deviation

**Step 1: Why the Z-distribution?**

This time the population standard deviation is **known** (σ = 0.20 million characters). When σ is known, the sampling distribution of the mean is Normal even for small samples, so we use the **Z-distribution** instead of the t-distribution — there is no need to estimate variability from the sample, so no extra uncertainty adjustment is required.

**Step 2: Formula for the Confidence Interval**

CI = x̄ ± Z(α/2) × (σ / √n)

where Z(α/2) is the critical Z-value for a 99% confidence level (α = 0.01).

In [2]:
# Python Code — Part (b)
sigma = 0.2   # known population standard deviation

z_crit = stats.norm.ppf(1 - alpha/2)
se_pop = sigma / np.sqrt(n)
margin_error_z = z_crit * se_pop

ci_lower_z = x_bar - margin_error_z
ci_upper_z = x_bar + margin_error_z

print("Population std dev (σ):", sigma)
print("Critical Z-value:", round(z_crit, 4))
print("Margin of Error:", round(margin_error_z, 4))
print("99% Confidence Interval (Z-distribution): (", round(ci_lower_z, 4), ",", round(ci_upper_z, 4), ")")

Population std dev (σ): 0.2
Critical Z-value: 2.5758
Margin of Error: 0.133
99% Confidence Interval (Z-distribution): ( 1.1057 , 1.3717 )


**Conclusion (Part b):**
Using the known population standard deviation of 0.20 million characters, we are 99% confident that the true mean durability lies between approximately **1.106 million and 1.372 million characters**.

**Comparing (a) and (b):** The Z-based interval is slightly narrower than the t-based interval, because the t-distribution's heavier tails (used when σ is unknown and estimated from a small sample) produce a larger critical value and hence a wider margin of error.